zebra-network/src/protocol/external/codec.rs

```diff
@@ -628,6 +628,9 @@ impl Codec {
     /// Currently, Zebra parses received `addrv2`s, ignoring some address types.
     /// Zebra never sends `addrv2` messages.
     pub(super) fn read_addrv2<R: Read>(&self, reader: R) -> Result<Message, Error> {
+        // TODO: Deserialize this field without `ZcashDeserialize`, check if there is some network id
+        //       in `self.with_testnet_parameters(NetworkParameters::network_id)`, and use that instead
+        //       of the default value. Check that serialization uses configured values too.
         let addrs: Vec<AddrV2> = reader.zcash_deserialize_into()?;
```

Comment:
This is incorrect and does not follow the specification in https://zips.z.cash/zip-0155